nucypher/config/node.py

```diff
@@ -617,21 +640,47 @@ def attach_keyring(self, checksum_address: str = None, *args, **kwargs) -> None:
                                        *args, **kwargs)
 
     def write_keyring(self, password: str, **generation_kwargs) -> NucypherKeyring:
-
+        # Note: It is assumed the blockchain is not yet available.
         if not self.federated_only and not self.checksum_public_address:
-            if self.geth:
-                data_dir = os.path.join(self.config_root, '.ethereum', NuCypherGethDevnetProcess._CHAIN_NAME)
-                etherbase = NuCypherGethDevnetProcess.ensure_account_exists(password=password, data_dir=data_dir)
 
-            checksum_address = self.blockchain.interface.w3.eth.accounts[0]  # etherbase
-        else:
+            #
+            # Integrated Provider Process
+            #
+
+            if self.provider_process:
+
+                if not os.path.exists(self.provider_process.data_dir):
+                    os.mkdir(self.provider_process.data_dir)
```

Comment:
is a `mode` needed?

Reply:
Hmm - Yeah explicit is better than implicit?